### PR TITLE
static-build: update libicu version

### PR DESCRIPTION
**Pull request by olegrok**
This patch updates libicu version to 71.1.
See changelog for details (https://icu.unicode.org/download/71).
The main reason to do that is a bug in datetime parsing that was
found using icu-date module (tarantool/icu-date#27)
and was fixed in linicu upstream
(https://unicode-org.atlassian.net/browse/ICU-21802).

NO_DOC=build
NO_TEST=build

### CODE REVIEW

**Review by locker — COMMENTED** [review]
LGTM (except the small nitpick regarding the changelog), but we need to fix the [`static_build_cmake_linux`](https://github.com/tarantool/tarantool/runs/5985761029?check_suite_focus=true) failure before we can merge it. I asked the DevX team, and they said that the job was killed by OOM. Probably, the failure is caused by the [unlimited number of jobs `make` is allowed to spawn](https://github.com/tarantool/tarantool/blob/d627ee2618c9f3e28cd91eab4b3f753379b5ce8c/.travis.mk#L274).

Also, in the future please don't create branches in the main repository. Fork the main repository instead and create a PR from your own fork.

**Comment by locker** [comment]
@olegrok #7020 was fixed. Please rebase. Let's see if it helps.

File: changelogs/unreleased/update-libicu-static-build.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+## feature/build
+
+* Update libicu version to 71.1 for static build.
```
[the remark this answers is not in the document]
Comment (olegrok) [reply changelogs/unreleased/update-libicu-static-build.md:3]: Fixed

**Comment by olegrok** [comment]
> LGTM (except the small nitpick regarding the changelog), but we need to fix the [`static_build_cmake_linux`](https://github.com/tarantool/tarantool/runs/5985761029?check_suite_focus=true) failure before we can merge it. I asked the DevX team, and they said that the job was killed by OOM. Probably, the failure is caused by the [unlimited number of jobs `make` is allowed to spawn](https://github.com/tarantool/tarantool/blob/d627ee2618c9f3e28cd91eab4b3f753379b5ce8c/.travis.mk#L274).

Seems after https://github.com/tarantool/tarantool/issues/7020 was fixed static_build_cmake_linux works fine.

> Also, in the future please don't create branches in the main repository. Fork the main repository instead and create a PR from your own fork.

Ok, no problems

**Review by locker — APPROVED** [review]

**Pull request merged**